Public/Src/Utilities/Storage/InputChange/InputChangeList.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using System.Linq;
+using BuildXL.Storage.ChangeTracking;
+using BuildXL.Storage.Tracing;
+using BuildXL.Utilities.Instrumentation.Common;
+
+namespace BuildXL.Storage.InputChange
+{
+    /// <summary>
+    /// Class representing input change list.
+    /// </summary>
+    public sealed class InputChangeList : IObservable<ChangedPathInfo>
+    {
+        private readonly List<IObserver<ChangedPathInfo>> m_observers = new List<IObserver<ChangedPathInfo>>(2);
+        private readonly List<ChangedPathInfo> m_changedPaths = new List<ChangedPathInfo>();
+        private static char[] s_inputSeparator = new[] { '|' };
+
+        /// <summary>
+        /// Gets the list of input changes.
+        /// </summary>
+        public IEnumerable<ChangedPathInfo> ChangedPaths => m_changedPaths;
+
+        private InputChangeList()
+        {
+        }
+
+        /// <summary>
+        /// Creates and instance of <see cref="InputChangeList"/> from file.
+        /// </summary>
+        public static InputChangeList CreateFromFile(LoggingContext loggingContext, string path)
+        {
+            Contract.Requires(loggingContext != null);
+            Contract.Requires(!string.IsNullOrEmpty(path));
+
+            try

```

Comment:
Please do an explict check for file existence and then log specific error about file not found and return null.
This eases debugging as it won't be a first chance exception.